### DISCUSSION TITLE
Show element uploads as Buttons with click event

### DISCUSSION THREAD

**Discussion by bsdRuf** [AvaloniaUI/Avalonia#11900] [Q&A]
Greetings AaloniaUI Githubers,

I've searched the API but I didnt find what I was looking for, may you can help me out?

I am getting started with Avalonia coming from WinForms and PHP/Html.
I am looking for UI manipulation like JS with DOM.

Whats the best way to let user add attachments and show them (interactive) in UI?

I am trying to let a user upload files (txt, jpg, png) and show them the list of the uploads as buttons (named like the file which has been uploaded).

The Plan: Each element should be a own button, click = remove

Thanks in advance

**Comment by timunie** [comment]
Best to start learning the MVVM aprroach, so the ViewModel can handle all of it. You don't usually add UI elements on your own in Avalonia. Check out the docs and https://github.com/AvaloniaUI/Avalonia.Samples

**Comment by bsdRuf** [comment]
I've checked this MVVM samples before. They are using the input text in a element that already exists.

I checked the https://docs.avaloniaui.net/guides/basics/code-behind but I also didnt got my answer on how to create a Avalonia.Controls.Button with event dynamicly.

How can I create a Avalonia.Controls.Button and add them to a Stackpanel by .cs code? My attempts to do so failed.

On the button "upload" I have this click event.
```

public System.EventHandler<Avalonia.Interactivity.RoutedEventArgs> Click;
//public event EventHandler<Avalonia.Interactivity.RoutedEventArgs> Click;

//public Avalonia.Interactivity.RoutedEvent? RemoveItemBtnClick(object? sender, Avalonia.Interactivity.RoutedEventArgs args)
public System.EventHandler<Avalonia.Interactivity.RoutedEvent?> RemoveItemBtnClick(object? sender, Avalonia.Interactivity.RoutedEventArgs args)
{
    return null;
}

public void AddAttachBtnClick(object? sender, Avalonia.Interactivity.RoutedEventArgs args)
{
    // file picker code, returning filename and file in a filestream

    // create new Button
    Button btn = new Button
    {
        Margin = new Avalonia.Thickness(5, 5, 5, 5),
        Foreground = (Avalonia.Media.IBrush)Brushes.Blue,
        Background = (Avalonia.Media.IBrush)Brushes.Transparent,
        BorderBrush = (Avalonia.Media.IBrush)Brushes.Transparent
    };
    btn.Content = "" + fileName;
    Click += new Avalonia.Interactivity.RoutedEventArgs(RemoveItemBtnClick(btn, this.Click ) );

    // Add button
    StackPanel spAttach = this.Find<StackPanel>("Attachments");
    spAttach.DataContext = btn;
}
```

**Reply by timunie** [reply]
you normally use DataTemplates which describes how your ViewModel or Model is rendered.
